version.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+var (
+	// BuildDate is date when binary was built
+	BuildDate string
+	// BuildVersion is the version of binary
+	BuildVersion string
+)
+
+// providerVersion holds current provider version
+type providerVersion struct {
+	Version   string `json:"version"`
+	BuildDate string `json:"buildDate"`
+	// MinDriverVersion is minimum driver version the provider works with
+	MinDriverVersion string `json:"minDriverVersion"`
+}
+
+func printVersion() (err error) {
```

Comment:
Consider passing in an `io.Writer`, which you can the `fmt.Fprintf` to. This is generally a useful decoupling, and will allow us to avoid all of the  special handling of Stdout in the test.

Reply:
returning `(string,error)`